### PR TITLE
Lint and validate YAML files in CI

### PR DESCRIPTION
**Pull request by sgibson91**
### Summary

After CI was broken due to malformed YAML in https://github.com/alan-turing-institute/the-turing-way/pull/2726, @aleesteele and I decided to implement linting of YAML files in CI.

### List of changes proposed in this PR (pull-request)

* Adds a new GitHub Actions workflow that is triggered on pushes to `*.yaml` and `*.yml` files
* This new workflow will run [`yamllint`](https://yamllint.readthedocs.io/en/stable/index.html) over the whole repo
* We used a relaxed config in order to minimise the diff, but this config _will_ catch the original issue in #2726 
* Updated YAML files in the repo to be properly formatted according to the relaxed config
* We explicitly ignore the `book/website/_config.yml` file as the comments are more readable when _not_ following pure YAML formatting

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by aleesteele** [comment]
Tagging @JimMadge and @da5nsy here, in case you have time to review! 🙏

File: .github/workflows/lorem-ipsums.yml [hunk 1]
```diff
@@ -25,28 +25,28 @@ jobs:
...
+        if: github.event.pull_request
+        run: |
+          python lorem-ipsums.py --pull-request ${{ github.event.pull_request.number }}
 
-    - name: Run Python script to check for rogue "Lorem ipsums" - Master
-      working-directory: ./tests
-      if: github.event_name == 'push' && github.ref == 'refs/heads/main'
-      run: |
-        python lorem-ipsums.py
+      - name: Run Python script to check for rogue "Lorem ipsums" - Master
```
Comment (JimMadge) [inline .github/workflows/lorem-ipsums.yml:48]:
```suggestion
      - name: Run Python script to check for rogue "Lorem ipsums" - Main branch
```

File: .github/workflows/no-bad-latin.yml [hunk 2]
```diff
@@ -25,28 +25,28 @@ jobs:
...
+        with:
+          python-version: 3.7
+      # Install Python dependencies
+      - name: Install dependencies
+        working-directory: ./tests
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt
+      # Run a Python script
+      - name: Run Python script to check for latin phrases - Master
```
Comment (JimMadge) [inline .github/workflows/no-bad-latin.yml:42]:
```suggestion
      - name: Run Python script to check for latin phrases - Main branch
```

**Review by JimMadge — CHANGES_REQUESTED** [review]
LGTM :+1:

There are a few references to 'master', the old default branch name, in the GitHub workflow files so it might be worth updating those while we are looking.

**Review by JimMadge — APPROVED** [review]

**Review by da5nsy — APPROVED** [review]
COOL! LGTM

**Review by aleesteele — APPROVED** [review]

**Pull request merged**